### PR TITLE
[RHOAIENG-90] - Add InstascaleMachinePool test

### PR DESCRIPTION
**Pull request by ChughShilpa**
[RHOAIENG-90](https://issues.redhat.com/browse/RHOAIENG-90)

### CODE REVIEW

File: ods_ci/tests/Tests/650__distributed_workloads/test-run-codeflare-tests.robot [hunk 1]
```diff
@@ -56,6 +57,43 @@ Run TestMnistPyTorchMCAD ODH test
...
+    ...                This particular test runs on OSD cluster
+    [Tags]  Instascale
+    ...     Tier3
+    ...     DistributedWorkloads
+    ...     CodeflareOperator
+    Skip If RHODS Is Self-Managed
+
+    # Generate ocm token and create a secret
+    Log To Console    "Generating token ....."
+    ${token} =    Run Process    ocm token --generate
```
Comment (sutaakar) [inline ods_ci/tests/Tests/650__distributed_workloads/test-run-codeflare-tests.robot:71]: This step expects user to be logged into OCM.
Is logging into OCM implicit for this test or does it have to be explicitly invoked?

File: ods_ci/tests/Tests/650__distributed_workloads/test-run-codeflare-tests.robot [hunk 2]
```diff
@@ -100,6 +138,8 @@ Run Codeflare E2E Test
     ...    cwd=${CODEFLARE_DIR}
     ...    env:CODEFLARE_TEST_TIMEOUT_LONG=30m
     ...    env:CODEFLARE_TEST_OUTPUT_DIR=%{WORKSPACE}/codeflare-e2e-logs
+    ...    env:CLUSTER_TYPE=OSD
```
Comment (sutaakar) [inline ods_ci/tests/Tests/650__distributed_workloads/test-run-codeflare-tests.robot:141]: Can you make a parameter out of this hardcoded value?

**Comment by ChughShilpa** [comment]
InstascaleMachinePool tests are failing with this error - https://opendatascience-jenkins-csb-rhods.apps.ocp-c1.prod.psi.redhat.com/job/distributed-workloads/job/rhods-pipeline-dw/120/console#:~:text=Unexpected%20error%3A,54%3A53%20%20%20%20%20%20%20%20%20%20occurred

These tests are passing locally in another OSD cluster (which is in different ocm org from ours qe) , can share the details if needed.

@aloganat @jgarciao Could you please take a look ? Thanks !!

File: ods_ci/tests/Tests/650__distributed_workloads/test-run-codeflare-tests.robot [hunk 3]
```diff
@@ -56,6 +57,43 @@ Run TestMnistPyTorchMCAD ODH test
...
+    Log To Console    "Setting instascale to true in config map ....."
+    ${result} =    Run Process    oc get cm codeflare-operator-config -n redhat-ods-applications -o yaml | sed -e 's|enabled: false|enabled: true|' | oc apply -f -
+    ...    shell=true    stderr=STDOUT
+    IF    ${result.rc} != 0
+        FAIL    Can not enable instascale to true
+    END
+
+    #Fetch cluster ID
+    Log To Console    "Fetching cluster ID ....."
+    ${cluster_id} =    Run Process    ocm list clusters | grep %{TEST_CLUSTER} | awk '{print $1}'
```
[the remark this answers is not in the document]
Comment (ChughShilpa) [reply ods_ci/tests/Tests/650__distributed_workloads/test-run-codeflare-tests.robot:90]: It is taken from jjenkns job's parameter

File: ods_ci/tests/Tests/650__distributed_workloads/test-run-codeflare-tests.robot (hunk 1 above)
Comment (ChughShilpa) [reply ods_ci/tests/Tests/650__distributed_workloads/test-run-codeflare-tests.robot:73]: OCM login is implicit

File: ods_ci/tests/Tests/650__distributed_workloads/test-run-codeflare-tests.robot (hunk 3 above)
[the remark this answers is not in the document]
Comment (sutaakar) [reply ods_ci/tests/Tests/650__distributed_workloads/test-run-codeflare-tests.robot:90]: Shouldn't it be declared in `*** Variables ***` section then?

**Comment by ChughShilpa** [comment]
Closed as the required jira is closed now - https://issues.redhat.com/browse/RHOAIENG-90

**Pull request closed without merging**